bears/xml2/XCopBear.py

```diff
@@ -0,0 +1,56 @@
+import itertools
+import re
+
+from coalib.bearlib.abstractions.Linter import linter
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+from dependency_management.requirements.GemRequirement import GemRequirement
+
+
+@linter(executable='xcop')
+class XCopBear:
+    """
+    This bear can fix formatting errors in your XML, XSD, XSL, XHTML file.
+    """
+    LANGUAGES = {'XML', 'XSD', 'XSL', 'XHTML'}
+    REQUIREMENTS = {GemRequirement('xcop')}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_FIX = {'Formatting'}
+    SEE_MORE = 'https://github.com/yegor256/xcop'
+
+    _output_regex = re.compile(
+        r'.*:(?P<line>\d+):.*(?P<severity>error|warning)\s?: '
+        r'(?P<message>.*)\n.*\n.*')
+
+    @staticmethod
+    def create_arguments(filename, file, config_file):
+        return filename,
+
+    def process_output(self, output, filename, file):
+        stdout = output
+        invalid_format = re.compile(r'\x1b\[31m.+?0m', re.S)
+        correct_formating = filename + ' looks good\n'
+
+        if(stdout != correct_formating):
+            """Output of Xcop is colour coded in the terminal to
+               identify the changes suggested by xcop.Green to indicate
+               addition and red to denote subtraction.
+
+               This introduces redundancies in the output file.
+               ^[[31m denoting subtraction (i.e red colour on terminal)
+               and ^[[32m denoting green.
+
+               Thus to remove these redundancies regex is used"""
+
+            stdout = re.sub(r'Invalid XML formatting in ' +
+                            filename+'\n', '', stdout)
+            stdout = re.sub(r'\\n', '', stdout)
+            stdout = re.sub(invalid_format, '', stdout)
+            stdout = re.sub(r'\x1b.+32m', '', stdout)
```

Comment:
this is begging for future breakages.  they can add more color, with different ansi sequences, and our bear breaks.

Use a de-ansi-ifier library , so it is done properly for this bear.

Also then create an issue in coala core to have de-ansi-ifying flag on the Linter in the core, so we can support many colorised output linters without custom logic for each.

Reply:
@Makman2 , maybe useful to have a way for a Linter to provide a stream cleaner/fixer, so that a simple regex can then be used for the cleaned stream.

Reply:
Sure will do that... It sounds great 👍

Reply:
@jayvdb  I used a deansifying library `https://pypi.python.org/pypi/ansicolors` which has an option to `strip_escapes` which removes all escape characters from a string. But the output of the linter is in such a way 

https://imgur.com/a/AeoFz

This creates an issue as the output is not only the corrected file in a colored format but rather a github-esque diff output.So if the color is stripped it's tough to find just the corrected output from it.

Reply:
If the output is a unified diff, `@linter` actually has native support for this.  See `unified-diff` in http://api.coala.io/en/latest/Developers/Writing_Linter_Bears.html

So if you do https://github.com/coala/coala/issues/5281 , then xcop bear will just use the de-ansi and unified-diff routine, and then probably doesnt need any custom logic.

Reply:
Or you could do https://github.com/yegor256/xcop/issues/19 , which is begging for someone to do it :P

Reply:
Also the functionality of `unified-diff` is `process_output_unified_diff`, which you could manually invoke in your bear after the de-ansi-ification.